### PR TITLE
Remove n+1 queries with includes

### PR DESCRIPTION
**Pull request by s-espinosa**
@srt32 This adds method to User to pull info needed for the serializers we use, then uses that method in the API users controllers.

Let me know what you think!

### CODE REVIEW

File: app/serializers/single_user_serializer.rb [hunk 1]
```diff
@@ -29,11 +29,11 @@ def cohort
   end
 
   def groups
-    object.groups.map {|group| group.name}
+    object.groups.pluck(:name)
```
Comment (srt32) [inline app/serializers/single_user_serializer.rb]: Does `pluck` not trigger a new db call?

**Comment by srt32** [comment]
LGTM after we confirm that `pluck` does indeed do an in memory call

**Comment by s-espinosa** [comment]
Tried this in the console and didn't see a second DB query. Looks like this behavior was also something that was changed last year specifically so that it wouldn't generate a new query ([link](https://github.com/rails/rails/issues/28747)). Looks like we're running Rails 5.2.1 on this app, so should be in the clear.

Going to go ahead and merge based on your comment above.

Related note: I think that one other piece that is currently slowing this down is the behavior of `#cohort` on user. It calls to `Cohort.find`, which calls to `Cohort.all`, which pulls all of the cohorts  and makes a new object for each one before running a `.find` to select the cohort with the correct id.

I think we could speed this up by making a call to Enroll for a specific Cohort's information, but it looks like the GraphQL implementation on Enroll doesn't yet have a resolver for that (I think? I'm at the edge of my knowledge with GraphQL).

Does it make sense to add that functionality to Enroll? In general the GraphQL implementation there seems pretty thin I think?

**Comment by srt32** [comment]
Good digging.

How slow is it?

We could add the single fetch endpoint on the Graphql API or we could cache the cohorts fetch and not worry about ;)

```
def cohorts
  Rails.cache.fetch('cohorts') do
    # slow query
  end
```

WDYT?

**Comment by s-espinosa** [comment]
Couple more questions on this:

1. Does this have continuous integration set up for deploy? It looks like it does in the `.circle_ci` directory, but wanted to double check. I'm still seeing the N+1 queries, and (if this is deployed) thinking that we maybe go back to `.map` to see if I was wrong about the database calls? If it doesn't have CI setup, let me know if this has already been deployed?

2. Re: caching - It seems like the time to respond varies a little bit. I tried to hit the endpoint in Census that just returned a list of Cohorts (and uses the Cohort.all call), and it was taking between 180ms and 400ms.

I'm a little on the fence. It seems if we have one API endpoint in Enroll then we should probably build it out to give us the functionality that we want, but I'm not positive what the discussion was for creating that endpoint in the first place. Do you have any more background?

    * Is there any reason not to build that out?
    * Is it something that we could use elsewhere, or was this built strictly for Census?

That said, maybe the answer is 'why not both?' I don't think it hurts to have the endpoint, I also don't think that Cohort data will change that often, so maybe the answer is to cache that response for a month or so? I think that they create the cohorts on Enroll quite a bit in advance, so I feel like we can keep that cache for quite a bit.

Thoughts?

**Comment by srt32** [comment]
re: 1, yes, auto deploy is set up via CI. Staging branch goes to staging. Master branch goes to production. Lemme know if that's not happening (you can look in heroku logs to confirm).

re: 2, within Turing apps, cohorts are meant to be fetched from census (census gets them from enroll due to legacy reasons). So,  I would not recommend we build out the Enroll API's much for cohorts but instead focus on getting Census API to be what we want it to be.

Happy to chat more!

**Pull request merged**